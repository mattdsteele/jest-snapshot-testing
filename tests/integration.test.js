const request = require('requisition');
const app = require('../src/app');
let server;

describe('web app', () => {
  beforeAll(async () => {
    server = app.listen('5000');
  });

  it('returns a response', async () => {
    const res = await request(`http://0.0.0.0:5000`);
    const body = await res.json();
    const { status, headers } = res;
    const snap = { status, body, headers };
    expect(snap).toMatchSnapshot();
  });

  afterAll(() => {
    server.close();
  });
});
