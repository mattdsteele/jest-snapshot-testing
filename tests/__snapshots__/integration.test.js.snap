// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web app returns a response 1`] = `
Object {
  "body": Array [
    Object {
      "number": "7.7.3",
      "url": "https://s3.amazonaws.com/heroku-nodejs-bins/node/release/linux-x64/node-v7.7.3-linux-x64.tar.gz",
    },
    Object {
      "number": "7.7.4",
      "url": "https://s3.amazonaws.com/heroku-nodejs-bins/node/release/linux-x64/node-v7.7.4-linux-x64.tar.gz",
    },
    Object {
      "number": "7.8.0",
      "url": "https://s3.amazonaws.com/heroku-nodejs-bins/node/release/linux-x64/node-v7.8.0-linux-x64.tar.gz",
    },
    Object {
      "number": "7.9.0",
      "url": "https://s3.amazonaws.com/heroku-nodejs-bins/node/release/linux-x64/node-v7.9.0-linux-x64.tar.gz",
    },
    Object {
      "number": "8.0.0",
      "url": "https://s3.amazonaws.com/heroku-nodejs-bins/node/release/linux-x64/node-v8.0.0-linux-x64.tar.gz",
    },
  ],
  "headers": Object {
    "connection": "close",
    "content-length": "616",
    "content-type": "application/json",
    "date": "Sat, 02 Nov 2019 19:37:46 GMT",
  },
  "status": 200,
}
`;
